pep-0612.rst

```diff
@@ -128,169 +156,346 @@ will defer the standardization of the semantics of those options to a later PEP.
 Valid use locations
 ^^^^^^^^^^^^^^^^^^^
 
-A declared ``ParameterSpecification`` can only be used in the place of the list
-of types in the declaration of a ``Callable`` type, or a user defined class
-which is generic in a ``ParameterSpecification`` variable (i.e., ``MyClass`` in
-the following example).
+Previously, the only things acceptable in the first "argument" to the
+``typing.Callable`` type are a list of parameter arguments (``[A, B, C]``) or an
+ellipsis (signifying "undefined parameters").  We now augment that with two new
+options:
+
+.. code-block::
+
+   parameters_expression ::=
+     | "..."
+     | "[" [ type_expression ("," type_expression)\* ] "]"
+     | parameter_specification_variable
+     | concatenate "["
+                      type_expression ("," type_expression)\* ","
+                      parameter_specification_variable
+                   "]"
+
+Where ``parameter_specification_variable`` resolves to a ``typing.ParamSpec``
+declaration as defined above, and ``concatenate`` resolves to
+``typing.type_variable_operators.Concatenate``.
+
+As before, ``parameters_expression`` is the only valid thing for the first
+parameter of ``typing.Callable``, but it is now acceptable in other user-defined
+generic types, as long as they are declared as being generic in
+a parameters_expression.
+
+.. code-block::
+
+   T = TypeVar("T")
+   S = TypeVar("S")
+   P_2 = ParamSpec("P_2")
+
+   class X(Generic[T, P]):
+     ...
+
+   def f(x: X[int, P_2]) -> str: ...                    # Accepted
+   def f(x: X[int, Concatenate[int, P_2]]) -> str: ...  # Accepted
+   def f(x: X[int, [int, bool]]) -> str: ...            # Accepted
```

Comment:
nit: this example doesn't seem to be super relevant. remove?

Reply:
Why is it not relevant?  Its the only way to spell a "literal" instantiation of a class parameteric on a paramspec

Reply:
Oh I see what you mean. It is not obvious to me though -- an explanation in the comment might be better but that's just me.

Reply:
I don't think its super super important since I don't imagine it will be used a ton, but I think its worth mentioning for refererence's sake.

Accordingly I think it's worth including but not dwelling on.